src/core/gosling-to-higlass.ts

```diff
@@ -120,6 +127,20 @@ export function goslingToHiGlass(
             };
         }
 
+        // We use higlass 'matrix' track instead of 'gosling-track' for rendering performance.
+        // HiGlass is really well-optimized for matrix visualization, and rendering it in Gosling
+        // makes the zooming interaction slow.
+        // See https://github.com/gosling-lang/gosling.js/pull/612#discussion_r771623844
+        if (IsHiGlassMatrix(firstResolvedSpec)) {
+            // By changing the track type, HiGlass uses its native heatmap track
+            hgTrack.type = 'heatmap';
```

Comment:
By replacing `'gosling-track'` w/ `'heatmap'`, HiGlass will handle this track for rendering and data managing, i.e., Gosling does not tabularize data needlessly.

Reply:
Great!